flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -119,6 +122,9 @@ public boolean equals(Object obj) {
 		}
 		else if (obj != null && obj.getClass() == ResourceProfile.class) {
 			ResourceProfile that = (ResourceProfile) obj;
+			if (this == ResourceProfile.UNIVERSAL || that == ResourceProfile.UNIVERSAL) {
+				return true;
+			}
```

Comment:
Here you're basically breaking the Java contract that `if a.equals(b), then a.hashCode() == b.hashCode()`